src/components/filters/filter_menu/filter_menu.ts

```diff
@@ -183,6 +181,40 @@ export class FilterMenu extends Component<Props, SpreadsheetChildEnv> {
     });
   }
 
+  /**
+   * Sorts an array of strings by treating them as either numeric values, dates, or strings.
+   *
+   * Numeric values are sorted in ascending order, while dates are sorted in chronological order.
+   * Strings are sorted based on the standard alphabetical order, with numeric and case sensitivity
+   * taken into account.
+   */
+  private sortValues(values: string[]): string[] {
+    const sortedValues = values.sort((a, b) => {
+      const aIsNumeric = !isNaN(+a);
+      const bIsNumeric = !isNaN(+b);
+      const aIsDate = !isNaN(Date.parse(a));
+      const bIsDate = !isNaN(Date.parse(b));
+
+      switch (true) {
+        case aIsNumeric && bIsNumeric:
+          return +a - +b;
+        case aIsNumeric:
+          return -1;
+        case bIsNumeric:
+          return 1;
+        case aIsDate && bIsDate:
+          return +new Date(a) - +new Date(b);
```

Comment:
It most probably won't work properly with other languages than English

Reply:
> It most probably won't work properly with other languages than English. I'm not sure there is a proper solution to the issue

Yes, that's correct. The `Date.parse()` method uses the English language conventions for parsing dates, and therefore it may not work correctly with input strings that use different languages. 

Maybe we can handle non-English date strings by converting them to English format before parsing them but I haven’t checked for its possibility yet.

On a second note, I guess we can ask PO how he thinks about this situation? What’s your opinion?

Reply:
> Maybe we can handle non-English date strings by converting them to English format before parsing them

Well "untranslate" the date strings would be hacky, unreliable, fragile, and can work only for some very specific cases we choose to address.

>  I guess we can ask PO how he thinks about this situation?

François agreed to keep the current behavior. (since he agreed to cancel the task last week)

Reply:
Just to be sure, by current behavior you're referring to `not sorting` the dates or is it this one with English only sorting? However, personally I will say letting it work only for English (Words & Numbers) are still better than nothing. Let me know what should we aim for?